sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -1492,7 +1492,7 @@ lsc_atomic_update(T *p, __ESIMD_NS::simd<uint32_t, N> offsets,
   __ESIMD_NS::simd<_MsgT, N> Tmp =
       __esimd_lsc_xatomic_stateless_1<_MsgT, _Op, L1H, L3H, _AddressScale,
                                       _ImmOffset, _DS, _VS, _Transposed, N>(
-          pred.data(), addrs.data(), src0.data());
+          pred.data(), addrs.data(), convert<_MsgT>(src0).data());
```

Comment:
- what is the type combination in the failing case?
- the corresponding test update does not seem to add any new test case

Reply:
Sorry, but I don't understand this change-set. If atomic for float value causes a compilation, then why convert is the proper solution?
Let's say the argument if float32 values 2.3, then convert to uint32_t would change 2.3 to 2.0 right and then to uint32_t value 2, right?

Reply:
> providing float as an arguments breaks the compilation

there is no legal user-level atomic API invocation which would translate into __esimd_lsc_xatomic_stateless_1 with floating-point argument. So I believe compilation failure is the right behavior.

Reply:
The additional fix replaced 'convert' with 'bit_cast'. Please answer the question asked above. How is that possible to have FP types in __esimd_lsc_xatomic_stateless_1()?
Why the bit_cast is needed here?

Reply:
Ok, this explanation helped. Thank you. 
So, fadd is 2 arguments, but here it is considered as 1 argument because the other/memory operand does not count and this function being updated here accepts fp operations like fadd correctly.